src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add a country to data base "
+                    + country.getName(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
```

Comment:
You don't need to set an ID for your entity explicitly. It'll be done by Hibernate. Ensure that the Country entity is configured properly with @GeneratedValue for the ID field.